File: src/main/java/com/bradhouse/differentiate/Nodes/VariableNode.java
package com.bradhouse.differentiate.Nodes;

/**
 * Created by brad on 6/7/16.
 */
public class VariableNode extends TreeNode {
    private int index;
    private TreeNode value;

    public VariableNode(int n) {
        index = n;
        value = null;
    }

    public void put(TreeNode t) {
        value = t;
    }

    public TreeNode get() {
        return value;
    }
}

File: src/main/java/com/bradhouse/differentiate/Nodes/VariableNode.kt
package com.bradhouse.differentiate.Nodes

class VariableNode(private val index: Int) : TreeNode() {
    private var value: TreeNode? = null

    override fun toString(): String {
        return "$" + index
    }

    fun put(t: TreeNode) {
        value = t
    }

    fun get(): TreeNode? {
        return value
    }
}
